core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -401,6 +407,31 @@ class GroupCoordinator(val brokerId: Int,
     }
   }
 
+  def handleListGroups(): (Errors, List[GroupOverview]) = {
+    if (!isActive.get) {
+      (Errors.GROUP_COORDINATOR_NOT_AVAILABLE, List[GroupOverview]())
+    } else {
+      (Errors.NONE, coordinatorMetadata.currentGroups.map(_.overview))
+    }
+  }
+
+  def handleDescribeGroup(groupId: String): (Errors, GroupSummary) = {
+    if (!isActive.get) {
+      (Errors.GROUP_COORDINATOR_NOT_AVAILABLE, GroupCoordinator.EmptyGroup)
+    } else if (!isCoordinatorForGroup(groupId)) {
+      (Errors.NOT_COORDINATOR_FOR_GROUP, GroupCoordinator.EmptyGroup)
+    } else {
+      val group = coordinatorMetadata.getGroup(groupId)
+      if (group == null) {
+        (Errors.NONE, GroupCoordinator.EmptyGroup)
```

Comment:
nit: Do we want to return an empty group with NONE error in this case? If we do then the client need to carefully handle the empty group case as you did now.